x-pack/plugins/fleet/server/routes/agent/upgrade_handler.ts

```diff
@@ -90,8 +99,9 @@ export const postBulkAgentsUpgradeHandler: RequestHandler<
   } = request.body;
   const kibanaVersion = appContextService.getKibanaVersion();
   try {
-    checkVersionIsSame(version, kibanaVersion);
+    checkKibanaVersion(version, kibanaVersion, false);
     checkSourceUriAllowed(sourceUri);
+    checkFleetServerVersion(version, agents, soClient, esClient);
```

Comment:
you are missing an `await` here `checkFleetServerVersion` is an async functiom

Reply:
Thanks for fixing it, I totally missed it